### PR TITLE
Use enum for DelayedConstructor<> so XCode can decode it in the debugger window...

### PR DESCRIPTION
**Pull request by mikelehen**

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/util/delayed_constructor.h [hunk 1]
```diff
@@ -121,7 +113,16 @@ class DelayedConstructor {
...
-  typename std::aligned_storage<sizeof(T), alignof(T)>::type space_;
+  union Space {
+    /** Default constructor does nothing. */
+    Space() {
+    }
+    ~Space() {
+      value.~T();
+    }
+    char empty;
+    T value;
```
Comment (wilhuff) [inline Firestore/core/src/firebase/firestore/util/delayed_constructor.h]: I can't comment on the places you haven't changed yet, but there are few other things that should go with this change:

  * All the placement `new` invocations in this file (`new (&space_) T`) should now reference `space_.value`.
  * All the `reinterpret_cast<...>(&space_)` can become just `space_.value`.

Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/util/delayed_constructor.h]: I've pushed a change containing these suggestions.

**Comment by mikelehen** [comment]
Thanks!  Will merge to firebase-master after CI runs.

**Comment by wilhuff** [comment]
Travis succeeded but didn't realize it. Merging.

**Pull request merged**